### PR TITLE
Add method to check whether Deferred Stats are present.

### PR DESCRIPTION
**Pull request by paul-r-gall**
Commit Message: Add method to check whether Deferred Stats are present.
Risk Level: low
Testing: unit
Docs Changes: none
Release Notes: none

### CODE REVIEW

**Review by adisuissa — COMMENTED** [review]
I don't see uses of these new methods, so I assume it will come later (and not something missing from this PR).
Assigning Josh for stats-related review.
/assign @jmarantz

File: envoy/stats/stats.h
Comment (jmarantz) [reply envoy/stats/stats.h:236]: +1 and also can you use doxgen syntax for the new one, and fix the one above it?

**Review by jmarantz — COMMENTED** [review]
lgtm modulo commenting nits.

we discussed use-case which is to scan over cluster stats accumulating them, without causing them to initialize if not already instantiated.

File: envoy/stats/stats.h
Comment (paul-r-gall) [reply envoy/stats/stats.h:246]: Done! Thanks for the review.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Please fix CI.
Assigning Stephen for !Google senior-maintainer review.
/assign @zuercher

**Review by jmarantz — APPROVED** [review]
@ggreenway for senior stamp

**Review by ggreenway — APPROVED** [review]

**Pull request merged**
